opentracing-spring-cloud-starter-jaeger-it/src/test/java/org/example/SpringWebApplicationTracesCollectedTest.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2017-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.example;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.concurrent.TimeUnit;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.LogMessageWaitStrategy;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = DemoSpringBootWebApplication.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ContextConfiguration(initializers = SpringWebApplicationTracesCollectedTest.Initializer.class)
+public class SpringWebApplicationTracesCollectedTest {
+
+
+  private static final int QUERY_PORT = 16686;
+  private static final int COLLECTOR_PORT = 14268;
+  private static final String SERVICE_NAME = "spring-boot-test";
+
+  @ClassRule
+  public static GenericContainer jaeger
+      = new GenericContainer("jaegertracing/all-in-one:1.3")
+      .withExposedPorts(COLLECTOR_PORT, QUERY_PORT)
+      //make sure we wait until the collector is ready
+      .waitingFor(new LogMessageWaitStrategy().withRegEx(".*jaeger-query.*HTTP server.*\n"));
+
+  @Autowired
+  private TestRestTemplate testRestTemplate;
```

Comment:
Why do you need two http clients? One would make it cleaner.

Reply:
The `TestRestTemplate` is used to interact with the actual Spring Boot app, while the `RestTemplate` is used to interact with the Jaeger server.
Using `TestRestTemplate` for both is probably technically possible, but I believe it would cause confusion due to it's explicit nature.